dkg/disk.go

```diff
@@ -127,58 +124,37 @@ func writeDepositData(aggSigs map[core.PubKey]*bls_sig.Signature, withdrawalAddr
 }
 
 // checkWrites writes sample files to check disk writes and removes sample files after verification.
-func checkWrites(dataDir string, def cluster.Definition) error {
-	var shares []share
-	sigs := make(map[core.PubKey]*bls_sig.Signature)
-	for i := 0; i < def.NumValidators; i++ {
-		tss, sks, err := tbls.GenerateTSS(def.Threshold, len(def.Operators), rand.Reader)
-		if err != nil {
-			return err
-		}
-
-		shares = append(shares, share{
-			PubKey:       tss.PublicKey(),
-			SecretShare:  sks[0],
-			PublicShares: tss.PublicShares(),
-		})
-
-		pk, err := tblsconv.KeyToCore(tss.PublicKey())
-		if err != nil {
-			return err
-		}
+func checkWrites(dataDir string) error {
+	checkBody := []byte("delete me: dummy file used to check write permissions")
 
-		sig, err := testutil.RandomBLSSignature()
-		if err != nil {
-			return err
-		}
-
-		sigs[pk] = sig
+	// Write all sample files and directories.
+	if err := os.WriteFile(path.Join(dataDir, "cluster-lock.json"), checkBody, 0o444); err != nil {
+		return errors.Wrap(err, "check writes")
 	}
 
-	if err := writeDepositData(sigs, testutil.RandomETHAddress(), "prater", dataDir); err != nil {
-		return err
+	if err := os.WriteFile(path.Join(dataDir, "deposit-data.json"), checkBody, 0o444); err != nil {
+		return errors.Wrap(err, "check writes")
 	}
 
-	if err := writeKeystores(dataDir, shares); err != nil {
-		return err
+	if err := os.MkdirAll(path.Join(dataDir, "validator_keys"), 0o444); err != nil {
+		return errors.Wrap(err, "check writes")
 	}
 
-	lock := cluster.Lock{Definition: def}
-	if err := writeLock(dataDir, lock); err != nil {
-		return err
+	if err := os.WriteFile(path.Join(dataDir, "keystore-0.json"), checkBody, 0o444); err != nil {
+		return errors.Wrap(err, "check writes")
 	}
 
-	// Cleanup sample files
-	if err := os.Remove(path.Join(dataDir, "deposit-data.json")); err != nil {
-		return errors.Wrap(err, "remove sample deposit-data.json")
+	// Remove all sample files and directories.
+	if err := os.RemoveAll(path.Join(dataDir, "validator_keys")); err != nil {
+		return errors.Wrap(err, "check writes")
 	}
 
-	if err := os.RemoveAll(path.Join(dataDir, "/validator_keys")); err != nil {
-		return errors.Wrap(err, "remove keys")
+	if err := os.Remove(path.Join(dataDir, "cluster-lock.json")); err != nil {
+		return errors.Wrap(os.Remove(path.Join(dataDir, "sample.txt")), "remove sample file")
```

Comment:
best not do fancy stuff when handling errors. Keep it simple, trying to do cleanup adds too much complexity. Also, you are dropping the first error

Reply:
yeah